platform-sdk/swirlds-base/src/main/java/com/swirlds/base/ArgumentUtils.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.base;
+
+/**
+ * Class that contains common checks like null checks as static methods.
+ */
+public class ArgumentUtils {
+
+    /**
+     * Private constructor to prevent instantiation.
+     */
+    private ArgumentUtils() {}
```

Comment:
```suggestion
    private ArgumentUtils() {
        throw new UnsupportedOperationException("not instantiable");
    }
```

Reply:
Already had that and removed it based on another review ;) @rbair23 can we add the pattern for private constructors to our code format? Should it be empty, throw an UnsupportedOperationException or an IllegalAccessException?

Reply:
I personally believe either way is fine; however, I agree with @qnswirlds the second one is safer if a piece of code with access to make the constructor accessible via reflection attempts to construct an object instance.

Reply:
But in that case IllegalAccessException would make more sense from my point of view.

Reply:
Added `IllegalAccessException` for now but open for any discussion :)